gateway-ha/src/main/java/com/lyft/data/gateway/ha/clustermonitor/ActiveClusterMonitor.java

```diff
@@ -134,6 +129,58 @@ private ClusterStats getPrestoClusterStats(ProxyBackendConfiguration backend) {
         conn.disconnect();
       }
     }
+    return null;
+  }
+
+  private ClusterStats getPrestoClusterStats(ProxyBackendConfiguration backend) {
+    ClusterStats clusterStats = new ClusterStats();
+    clusterStats.setClusterId(backend.getName());
+
+    // Fetch Cluster level Stats.
+    String target = backend.getProxyTo() + UI_API_STATS_PATH;
+    String response = queryCluster(target);
+    if (response == null) {
+      log.error("Received null response for {}", target);
+      return  clusterStats;
+    }
+    clusterStats.setHealthy(true);
+    try {
+      HashMap<String, Object> result = null;
+      result = OBJECT_MAPPER.readValue(response, HashMap.class);
+
+      clusterStats.setNumWorkerNodes((int) result.get("activeWorkers"));
+      clusterStats.setQueuedQueryCount((int) result.get("queuedQueries"));
+      clusterStats.setRunningQueryCount((int) result.get("runningQueries"));
+      clusterStats.setBlockedQueryCount((int) result.get("blockedQueries"));
+      clusterStats.setProxyTo(backend.getProxyTo());
+      clusterStats.setExternalUrl(backend.getExternalUrl());
+      clusterStats.setRoutingGroup(backend.getRoutingGroup());
+
+    } catch (Exception e) {
+      log.error("Error parsing cluster stats from [{}]", response, e);
+    }
+
+    // Fetch User Level Stats.
```

Comment:
this block makes sense to me. However, I am a little concerned when we have 200K+ adhoc query requests per day, which will lead to that amount of queued list transactions and API requests to coordinators. In the future we may want to look for a caching solution to refresh and memorize cluster status every once a while

Reply:
The query count doesn't matter for the number of API requests as we do this periodically and not on a per query basis.  Currently, we query clusters every minute i.e. 1 call/minute/cluster so routingManager is already using cached data till the ActiveClusterMonitor queries the clusters again after 1min. Also this is configurable so we can slow it further down if need be. 

Given we only fetch queued queries data, I don't think the amount of data will be a concern either given a typical 20-30 clusters setup with ~ 1000 concurrent queued queries per cluster.